File: Medium-Hard/DecimalToOctal.c
#include<stdio.h>

// void main(){
//     int n = 254;

//     int oct = 0;
//     int place = 1;

//     while (n > 0)
//     {
//         int rem = n % 8;
//         oct += rem * place;
//         place *= 10;
//         n /= 8;
//     }

//     printf("%d", oct);
// }

void main(){
    int n = 254;

    char octalDigits[] = "01234567";
    char octal[20];
    int index = 0;

    while (n > 0)
    {
        int rem = n % 8;
        octal[index] = octalDigits[rem];
        index++;
        n /= 8;
    }

    octal[index] = '\0';

    // Print in reverse order
    for (int i = index - 1; i >= 0; i--)
    {
        printf("%c", octal[i]);
    }
    
}

File: Medium-Hard/DecimalToOctal.java
public class DecimalToOctal {

    public static int decimalToOctal(int n){
        String octValues = "01234567";
        StringBuilder octal = new StringBuilder();

        while (n > 0) {
            int rem = n % 8;
            octal.append(octValues.charAt(rem));
            n /= 8;
        }

        return Integer.parseInt(octal.reverse().toString());
    }

    public static int octalToDecimal(int n){
        int decimal = 0;
        int power = 0;

        while (n > 0) {
            int rem = n % 10;
            decimal += rem * Math.pow(8, power);
            power++;
            n /= 10;
        }

        return decimal;
    }

    public static void main(String[] args) {
        
    }
}
